### PR TITLE
Moving from GSF to Conscrypt

### PR DESCRIPTION
**Pull request by kaeptmblaubaer1000**
This replaces #287.

### CODE REVIEW

**Comment by kaeptmblaubaer1000** [comment]
Travis still seems to have issues.

**Comment by kaeptmblaubaer1000** [comment]
@maks please trigger a rebuild

**Comment by kaeptmblaubaer1000** [comment]
Just the travis build failed because of jcenter, so don't trigger a rebuild.
Tested with Android 4.1.2, will test with 4.4 now.

**Comment by kaeptmblaubaer1000** [comment]
Works too on API 19, KitKat 4.4.

**Comment by maks** [comment]
Excellent! Thank you very much cleaning it up and redoing this PR.
Looks good and I'll look into whats happening on Travis.

File: app/src/main/java/com/manichord/mgit/transport/SSLProviderInstaller.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+                case -1:
+                    break;
+                default:
+                    String str = "Failed to install security provider MGit_OpenSSL, result: ";
+                    Timber.e(str + pos);
+                    throw new SecurityException();
+            }
+
+        } catch (NoSuchAlgorithmException e) {
+            Timber.e("Failed to find SSLContext.Default provider");
```
Comment (maks) [inline app/src/main/java/com/manichord/mgit/transport/SSLProviderInstaller.java]: For future PRs, please include the exception object when logging via Timber.e()

**Comment by maks** [comment]
@kaeptmblaubaer1000 thanks again for doing this!

File: app/src/main/java/com/manichord/mgit/transport/SSLProviderInstaller.java (hunk 1 above)
Comment (kaeptmblaubaer1000) [reply app/src/main/java/com/manichord/mgit/transport/SSLProviderInstaller.java]: @maks should I do this and create a PR?

**Pull request merged**